src/symbols/Edge.tsx

```diff
@@ -114,36 +116,73 @@ export const Edge: FC<EdgeProps> = ({
   onPointerOut
 }) => {
   const theme = useStore(state => state.theme);
-  const edge = useStore(state => state.edges.find(e => e.id === id));
+  const edges = useStore(state => state.edges);
+  const edge = edges.find(e => e.id === id);
+  const edgesWithSameSourceAndTarget = edges
+    .filter(e => e.target === edge.target && e.source === edge.source)
+    .map(e => e.id);
   const { target, source, label, labelVisible = false, size = 1 } = edge;
-  const curved = interpolation === 'curved';
+  let curved = interpolation === 'curved';
 
   const from = useStore(store => store.nodes.find(node => node.id === source));
   const to = useStore(store => store.nodes.find(node => node.id === target));
   const draggingId = useStore(state => state.draggingId);
   const [active, setActive] = useState<boolean>(false);
   const [menuVisible, setMenuVisible] = useState<boolean>(false);
-  const labelOffset = (size + theme.edge.label.fontSize) / 2;
+  let labelOffset = (size + theme.edge.label.fontSize) / 2;
+  let curveOffset: number;
 
   const [arrowLength, arrowSize] = useMemo(() => getArrowSize(size), [size]);
 
+  if (edgesWithSameSourceAndTarget.length > 1) {
```

Comment:
This needs some comments - I might even move it to a pure function for easier isolation.